Makefile

```diff
@@ -128,20 +129,26 @@ endif
 ifeq ($(RPM_REPOSITORY),oss)
 	mock -r inmanta-and-epel-7-x86_64 --bootstrap-chroot --enablerepo="inmanta-oss-$(RELEASE)" --buildsrpm --spec inmanta.spec --sources dist --resultdir ${RPMDIR-EL7}
 	mock -r inmanta-and-epel-7-x86_64 --bootstrap-chroot --enablerepo="inmanta-oss-$(RELEASE)" --rebuild ${RPMDIR-EL7}/python3-inmanta-${VERSION}-*.src.rpm --resultdir ${RPMDIR-EL7}
+
+	mock -r epel-8-x86_64 --bootstrap-chroot --enablerepo="inmanta-oss-$(RELEASE)" --buildsrpm --spec inmanta.spec --sources dist --resultdir ${RPMDIR-EL8}
```

Comment:
The `--enablerepo` should point to the staging repository when a new stable release is built. Otherwise the RPM is built against the wrong RPM dependency.

Reply:
I'm not sure I agree. For the ISO releases, we publish all components to and build them against the stable-staging repo, then promote them all at once. However, the `inmanta` package is just for backwards compatibility: `inmanta-core` is a fully self-sufficient package in all regards. So why should we have its promotion depend on the release of `inmanta`?

In any case, looking at the `inmanta-core` pipeline, it currently promotes the packages at the end of the pipeline, so if we conclude we want to delay that promotion until `inmanta` has been built and published to stable-staging, we should update that pipeline as well.

Reply:
(I've made the changes locally but I'll wait to push them until we've reached a decision here).

Reply:
It's important to build against the stable-staging RPM repository. The RPM build for `python3-inmanta` works as follows:

1. The `/opt/inmanta` venv is created by installing the `python3-inmanta-core` RPM as a dependency (See: BuildRequires)
2. Then, the `inmanta` package is installed in the venv
3. Finally, this line (https://github.com/inmanta/inmanta/blob/93e028f22eb6047a3354afd79982d337540927f0/inmanta.spec#L57) ensures that the resulting `python3-inmanta` RPM only contains the venv files which are not part of the `python3-inmanta-core` RPM. 

If we would build the `python3-inmanta` RPM against the `python3-inmanta-core` RPM of the previous release (against the stable repo) and the `python3-inmanta-core` RPM of the new release (on the stable-staging repo) contains an additional dependency, then this dependency will be packaged in two RPMs of the new release. Namely, the `python3-inmanta` and the `python3-inmanta-core` RPM. Because the new dependency was not installed in the inmanta venv via the BuildRequires when the `python3-inmanta` RPM was built. Installing that new `python3-inmanta` RPM would result in a file conflict, between the `python3-inmanta-core` and the `python3-inmanta` RPM.

Reply:
The RPM build for `python3-inmanta` works as follows:

1. The `/opt/inmanta` venv is created by installing the `python3-inmanta-core` RPM as a dependency (See: BuildRequires)
2. Then, the `inmanta` package is installed in the venv
3. Finally, this line (https://github.com/inmanta/inmanta/blob/93e028f22eb6047a3354afd79982d337540927f0/inmanta.spec#L57) ensures that the resulting `python3-inmanta` RPM only contains the venv files which are not part of the `python3-inmanta-core` RPM. 

If we would build the `python3-inmanta` RPM against the `python3-inmanta-core` RPM of the previous release (against the stable repo), then we can never be sure that the resulting venv, i.e. the venv you get when installing the `python3-inmanta` and the `python3-inmanta-core` RPM, will be correct. I don't think this causes a problem for this specific build. But sooner or later this might cause problems...

Reply:
But python3-inmanta's version is pinned, isn't it? This means the repo shouldn't matter as long as it contains the correct version of the core package.

Reply:
Yes. But the version pinning is something that is relevant when you install the RPM. The problem I am talking about handles about the build of the RPM. Each RPM builds on top of another Inmanta RPM (except for the `python3-inmanta-core` RPM). Each new RPM extends the Inmanta venv a bit. As such, the order in which the RPMs are built is important, because each additional RPM needs to know which files are already present in the Inmanta venv up to now. This also means that it's important that each RPM is built against an RPM of the same release and not the RPM of a previous release. Otherwise the RPM built has an incorrect view on which files the other RPMs already install in the Inmanta venv. Clear?

Reply:
I fully agree on everything you say, except for the conclusion, I still don't see why the repo should matter. `BuildRequires` pinning should make sure we build on top of the correct version, shouldn't it?

Reply:
But that version will not be present in the stable repo. It will only be present the stable-staging repo, right?

Reply:
In the current implementation of the inmanta-core build pipeline, the package gets promoted immediately, so it should be present in the stable repo by the time we execute this build job.
(I'm not sure how relevant this discussion is now. We'll move away from this soon but on the other hand I don't think we want to leave this in an inconsistent state so we probably need to consolidate this anyway?)

Reply:
Then we have a but in the pipeline. The promote from stable-staging to stable should only happen as soon as all RPMs are present. The goal of the staging repo is to make all the RPMs available in the stable repo in an atomic way.

Reply:
For ISO yes, but that doesn't necessarily have to be the case for OSS if `inmanta-core` is a standalone package. Now that we'll introduce the two product RPMs I guess that won't be the case anymore though. Either way, with the work we're planning to do this sprint I think this discussion is mostly moot. So how do we proceed in the meantime?
1. leave inmanta-core pipeline and this PR as is, with what could be argued to be a bug but won't actually cause any issues because inmanta-core is self-sufficient.|
2. update the inmanta-core pipeline to not promote any packages and push my changes to this PR so that it enables the staging repo.
I'll leave the decision up to you.

Reply:
All this code will be rewritten so it doesn't make sense to spend any more time on this. You can leave it as it is.